docs/snippets/src/main.py

```diff
@@ -0,0 +1,52 @@
+# Copyright (C) 2018-2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+#! [import]
+import openvino.runtime as ov
+#! [import]
+
+#! [part1]
+core = ov.Core()
+model = ov.Model()
+compiled_model = ov.CompiledModel()
+#! [part1]
+
+#! [part2_1]
+compiled_model = core.compile_model("model.xml", "AUTO")
+#! [part2_1]
+#! [part2_2]
+compiled_model = core.compile_model("model.onnx", "AUTO")
+#! [part2_2]
+#! [part2_3]
+compiled_model = core.compile_model("model.pdmodel", "AUTO")
+#! [part2_3]
+#! [part2_4]
+compiled_model = core.compile_model(model, "AUTO")
+#! [part2_4]
+
+#! [part3]
+infer_request = compiled_model.create_infer_request()
+#! [part3]
+
+memory = np.array([1, 2, 3, 4])
+#! [part4]
+# Get input port for model with one input
+input_port = model.input();
+# Create tensor from external memory
+input_tensor = ov.Tensor(array=memory, shared_memory=True)
+# Set input tensor for model with one input
+infer_request.set_input_tensor(input_tensor)
```

Comment:
Personally I would like to see a case where both modes for `shared_memory` are used and how it can affect inference. @ilyachur what do you think? Can this be done here?

Also docs can show how to use `set_input_tensor` with port obtained few lines before and show the "shortcut approach" without information of given port (one input models).

CC @akuporos

Reply:
I added a new dedicated page for extended cases.